include/ast.h

```diff
@@ -40,6 +40,8 @@ typedef struct s_redirect_info
 	t_redirect_type	type;
 	const char		*filepath;
```

Comment:
suggestion to use union

```suggestion
    union
    {
        const char *filepath;
        const char *delimiter;
    } file_or_delim;
```